modules/db_instance/main.tf

```diff
@@ -7,6 +7,7 @@ resource "aws_iam_role" "enhanced_monitoring" {
 
   name               = "${var.monitoring_role_name}"
   assume_role_policy = "${file("${path.module}/policy/enhancedmonitoring.json")}"
+  tags               = "${merge(var.tags, map("Name", format("%s", var.monitoring_role_name)))}"
```

Comment:
`"${merge(map("Name", var.monitoring_role_name), var.tags)}"` - use this to allow users to override default `Name` tag if necessary.

Also, add an empty line before `tags` argument.